ledger/apply/mockBalances.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package apply
+
+import (
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions/logic"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+type mockBalances struct {
```

Comment:
Could you add some comments describing how to use this tool?

Also, you moved it out of the `test` package, do we have some sort of testutils spot to put things like this instead of putting test code into release packages?

Reply:
Not sure about the "tool" side of it. It mocks the "real" balances which are well documented. 

I didn't. It was in `payment_test.go` and used by the other test files in the `apply` directory.  - I mainly put it in its own file and made it a bit more generic that it would be easier to use and clear that it doesn't belong to a specific test suite.

Reply:
I think that I agree with @winder here. it's perfectly fine to move it to it's own file, but unless we're using it from outside the package, we should keep it as a _test.go file.